core/sysparam.c

```diff
@@ -119,50 +122,28 @@ static struct {
 
 /***************************** Internal routines *****************************/
 
-static inline IRAM sysparam_status_t _do_write(uint32_t addr, const void *data, size_t data_size) {
-    CHECK_FLASH_OP(sdk_spi_flash_write(addr, (void*) data, data_size));
-    return SYSPARAM_OK;
-}
-
-static inline IRAM sysparam_status_t _do_verify(uint32_t addr, const void *data, void *buffer, size_t len) {
-    CHECK_FLASH_OP(sdk_spi_flash_read(addr, buffer, len));
-    if (memcmp(data, buffer, len)) {
-        return SYSPARAM_ERR_IO;
+static sysparam_status_t _write_and_verify(uint32_t addr, const void *data, size_t data_size) {
+    static uint8_t bounce[BOUNCE_BUFFER_SIZE];
```

Comment:
This looks much cleaner using the improved flash support, thanks. Could we move all the flash read/write paths across to the new code now and remove the sdk versions (in follow up work)?